### DISCUSSION TITLE
Problema com autenticação com o Azure AD - aad_oauth 0.4.0

### DISCUSSION THREAD

**Discussion by morgana-basso** [Flutterando/forum#51] [Perguntas e Respostas] [answered]
Boa tarde pessoal,

Vou descrever uma dificuldade que estou passando, caso alguém possa me ajudar com alguma experiência que já tenha passado também!

Estou com problema em utilizar a package aad_oauth 0.4.0 para autenticação com o Azure AD
E no projeto eu utilizo o flutter_modular 5.0.3, que acredito ser algo nisso que talvez esteja faltando alguma configuração.
Pois testando em um projeto sem o flutter modular, a package aad_oauth funciona corretamente.

O erro acontece neste comando 
await oauth.login();
-> Null check operator used on a null value

E eu penso que seja algo relacionado ao navigatorKey que foi a última atualização do aad_oauth

Tentei setar ele no meu app_widget, mas ainda assim não funcionou.
![image](https://user-images.githubusercontent.com/73257278/194626749-a47f63f4-a8dc-40a0-a462-09a25e198b87.png)

**Comment by eduardoflorence** [comment]
@morgana-basso tem certeza que o erro acontece nesta linha do await?
O erro "Null check operator used on a null value" acontece quando usamos a exclamação para forçar que o compilador entenda o valor como não nulo, numa situação que ele possa ser nulo, so que ele não é. Exemplo:
```dart
final a = null;
final a!.login()  // aqui daria o erro por causa do ponto de exclamação
```

Como você está determinando o valor de `oauth`?

**Reply by morgana-basso** [reply]
Olá!

Eu acredito que o problema não esteja diretamente relacionada a linha await oauth.login(), apesar do erro ocorrer aqui.
Acredito que esteja relacionado ao navigatorKey, que na documentação diz que deve ser passado no MaterialApp().

Este é um projeto de teste que está funcionando, mas que toda esta estrutura foi feita na mesma página.
O ponto que acredito ser o problema no meu outro projeto é o setNavigatorKey conforme vemos na linha 13.
![image](https://user-images.githubusercontent.com/73257278/194649623-e7dee53c-c5f4-46e4-9e8b-9dccfef5ac2f.png)


No meu projeto que está com problema, eu tenho separado por módulo as telas, e acho que está perdendo digamos a referência do navigatorKey.
Aqui eu tenho meu app_widget
![image](https://user-images.githubusercontent.com/73257278/194650207-b4b9662e-55e0-4d16-ad8d-be6186d9c70a.png)

E tenho separado o módulo do login, onde de fato utilizo a package para a autenticação
![image](https://user-images.githubusercontent.com/73257278/194651209-9f3dcd24-017b-4554-96ff-06352acacfe6.png)

**Reply by eduardoflorence** [reply]
@morgana-basso coloca o `final navigatorKey = GlobalKey<NavigatorState>;` apenas uma vez no seu código, acima do método main principal do seu app. Assim navigatorKey vira uma variável global e você pode utilizar em várias partes do seu código.

**Reply by morgana-basso** [reply]
Perfeito @eduardoflorence muito obrigada pela ajuda, consegui resolver colocando em um arquivo de constantes para utilizar em todas as partes do meu código.

**Comment by morgana-basso** [comment, marked as answer]
Acredito ter encontrado a solução:

inicializar o GlobalKey em outro arquivo, de constantes por exemplo:
![image](https://user-images.githubusercontent.com/73257278/194658470-488c602a-08d0-4ebc-a989-6ed0b43dcfc1.png)

E depois utilizando no main e login:

![image](https://user-images.githubusercontent.com/73257278/194659021-023248f7-28d6-42d2-9a10-69dfa40387b8.png)

![image](https://user-images.githubusercontent.com/73257278/194659286-1ca5c8f9-a508-4def-8007-203211075796.png)
